scripts/mod_loader/ui/widgets/root.lua

```diff
@@ -97,66 +183,50 @@ function UiRoot:event(eventloop)
 
 	if type == sdl.events.mousebuttondown then
 		local button = eventloop:mousebutton()
-		self:setfocus(nil)
-		local done = self:mousedown(mx, my, button)
-		if self:dropdownEvent(mx, my) then
-			done = self.currentDropDown:mousedown(mx, my, button) or done
-			table.remove(self.children)
+		local consumeEvent = false
+
+		self:releasePressedchild(mx, my, button)
+		self:setfocus(self.hoveredchild)
+
+		if button == 1 then
+			consumeEvent = self:pressHoveredchild(mx, my, button)
+		end
+
+		-- inform open dropDownUi's of mouse down event,
+		-- even if the mouse click was outside of its area,
+		-- in order to allow them to close
+		for _, dropdown in ipairs(self.dropdownUi.children) do
+			if not dropdown.containsMouse then
+				dropdown:mousedown(mx, my, button)
+			end
 		end
-		return done
+
+		self:cleanupDropdown()
+
+		return consumeEvent
 	end
 	
 	if type == sdl.events.mousebuttonup then
 		local button = eventloop:mousebutton()
-		local child = self.pressedchild
-
-		local dEvent = self:dropdownEvent(mx, my)
-		if
-			self.currentDropDown and self.currentDropDownOwner ~= child and
-			not dEvent
-		then
-			-- destroy the dropdown if we click somewhere away from it
-			self.currentDropDownOwner.hovered = false
-			self.currentDropDownOwner:destroyDropDown()
-		end
-		if dEvent then table.remove(self.children) end
-		
-		-- Notify pressed children of the event, even if the mouse is released
-		-- outside of them.
-		if self.pressedchild and self.pressedchild:mouseup(mx, my, button) then
-			self.pressedchild.pressed = false
-			self.pressedchild = nil
-			return true
+
+		if button == 1 and self.pressedchild ~= nil then
+			return self:releasePressedchild(mx, my, button)
+		elseif button == 3 and self.hoveredchild ~= nil then
```

Comment:
On further consideration; this elseif block is probably not the right call.
No system in the mod loader currently require to have their `mouseup` called on button 3 in order to consume the event before any of its ancestors have had their `mouseup` called.
I am considering removing it, so only mouse 1 will have a special case associated with it, and all other buttons (2-?) get their `mouseup` calls through the normal chain of `mouseup` calls via their ancestors. A `hoveredchild` should receive it as long as its ancestors haven't overridden their `mouseup` functions to prevent it. Which also doesn't seem unreasonable.

Reply:
Sounds good, it doesn't seem like it'll break anything as far as I can tell.